src/modules/users/guards/jwt-auth.guard.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { JwtService } from '@nestjs/jwt';
+import { ExecutionContext } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class JwtAuthGuard extends AuthGuard('jwt') {
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly configService: ConfigService,
+  ) {
+    super();
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest();
+    const token = this.extractTokenFromHeader(request);
+
+    if (!token) {
+      throw new UnauthorizedException('Token not provided');
+    }
+
+    try {
+      const decoded = await this.jwtService.verifyAsync(token, {
```

Comment:
passport has built-in methods for verifying tokens. you should look at how Anvar did his refresh token strategy and do in a similar way.

Reply:
i use built in library jwtservice

Reply:
i know that you are using built-in jwt service. but there are cleaner ways of doing this with passport. 

https://docs.nestjs.com/recipes/passport#implementing-passport-jwt

Anvar used a similar way like in the documentation. checkout his pr and his refresh token strategy and guard.